salt/utils/vault.py

```diff
@@ -126,24 +126,28 @@ def _get_vault_connection():
         return _get_token_and_url_from_master()
 
 
-def make_request(method, resource, profile=None, **args):
+def make_request(method, resource, token=None, vault_url=None, get_token_url=False, profile=None, **args):
     '''
     Make a request to Vault
     '''
     if profile is not None and profile.keys().remove('driver') is not None:
         # Deprecated code path
         return make_request_with_profile(method, resource, profile, **args)
 
-    connection = _get_vault_connection()
-    token, vault_url = connection['token'], connection['url']
-    if 'verify' not in args:
-        args['verify'] = connection['verify']
+    if not token or not vault_url:
```

Comment:
What is the expected behavior here if one is passed in, but not the other? If you pass in a token, but not a vault_url, the passed in token will be overridden, which could be confusing. Is that expected?

Reply:
@rallytime The expected behavior from the code is to re-auth but I don't like that much either, if for example the url is not given.
It would have been better to have the vault url by other means, but I'm not sure how. The utils is stateless.
_get_token_and_url_from_master()  returns the token along with the url (which it gets from the master via the peer run). Would it be better to have a function in the vault runner that just returns the vault url?

Reply:
Hrm. What do you think @gtmanfred? I think you've been involved in vault functionality lately.

Reply:
I am ok with this.  One of the things that sdb/vault should be able to do is reduce load on the master, and allow the minion to just look up secrets on its own.  And it would need to keep track of its token in order to do that.

But I would like to reserve the right to rip it out if it turns out to break vault + sdb in Fluorine.

I have made this issue to add tests before the Fluorine release https://github.com/saltstack/salt/issues/46052

Reply:
Wonderful! Thanks @gtmanfred :)

Reply:
That sound OK to you @iustinam, before we merge this in?